exercises/grep/tests/grep.rs

```diff
@@ -0,0 +1,561 @@
+Of that Forbidden Tree, whose mortal tast
+Brought Death into the World, and all our woe,
+With loss of Eden, till one greater Man
+Restore us, and regain the blissful Seat,
+Sing Heav'nly Muse, that on the secret top
+Of Oreb, or of Sinai, didst inspire
+That Shepherd, who first taught the chosen Seed
+";
+
+/// In The White Night
+/// A poem by Alexander Blok(https://en.wikipedia.org/wiki/Alexander_Blok)
+/// a Russian poet who is regarded as one of the most important figures of the Silver Age of Russian Poetry
+/// You can read the translation here: https://lyricstranslate.com/ru/белой-ночью-месяц-красный-white-night-crimson-crescent.html
+static IN_THE_WHITE_NIGHT_CONTENT: &'static str = "Белой ночью месяц красный
+Выплывает в синеве.
+Бродит призрачно-прекрасный,
+Отражается в Неве.
+Мне провидится и снится
+Исполпенье тайных дум.
+В вас ли доброе таится,
+Красный месяц, тихий шум?..
+";
+
+struct Fixture<'a> {
+    file_names: &'a [&'a str],
+}
+
+impl<'a> Fixture<'a> {
+    fn new(file_names: &'a [&'a str]) -> Self {
+        Fixture { file_names }
+    }
+
+    fn set_up(&self) {
+        let file_name_content_pairs = self.file_names
+            .iter()
+            .cloned()
+            .map(|file_name| {
+                if file_name.ends_with("iliad.txt") {
+                    (file_name, ILIAD_CONTENT)
+                } else if file_name.ends_with("midsummer_night.txt") {
+                    (file_name, MIDSUMMER_NIGHT_CONTENT)
+                } else if file_name.ends_with("paradise_lost.txt") {
+                    (file_name, PARADISE_LOST_CONTENT)
+                } else {
+                    (file_name, IN_THE_WHITE_NIGHT_CONTENT)
+                }
+            })
+            .collect::<Vec<(&str, &str)>>();
+
+        set_up_files(&file_name_content_pairs);
+    }
+}
+
+impl<'a> Drop for Fixture<'a> {
+    fn drop(&mut self) {
+        tear_down_files(self.file_names);
+    }
+}
+
+fn set_up_files(files: &[(&str, &str)]) {
+    for (file_name, file_content) in files {
+        fs::write(file_name, file_content).expect(&format!(
+            "Error setting up file '{}' with the following content:\n{}",
+            file_name, file_content
+        ));
+    }
+}
+
+fn tear_down_files(files: &[&str]) {
+    for file_name in files {
+        fs::remove_file(file_name).expect(&format!("Could not delete file '{}'", file_name));
+    }
+}
+
+fn process_grep_case(pattern: &str, flags: &[&str], files: &[&str], expected: &[&str]) {
+    let test_fixture = Fixture::new(files);
+
+    test_fixture.set_up();
+
+    let flags = Flags::new(flags);
+
+    let grep_result = grep(pattern, &flags, files).unwrap();
+
+    assert_eq!(grep_result, expected);
+}
+
+#[test]
+fn test_nonexistent_file_returns_error() {
+    let pattern = "Agamemnon";
+
+    let flags = Flags::new(&vec![]);
+
+    let files = vec!["test_nonexistent_file_returns_error_iliad.txt"];
```

Comment:
It is tedious and error-prone to manually specify the file names. You've already done the work, and I presume that this works, but you may want to replace this sort of thing with a macro which joins the test name with each filename. [Playground example](http://play.rust-lang.org/?gist=a983d9e6cae019d8a4f3abc643a2e00d&version=stable&mode=debug&edition=2015).